src/ts/ui/MainUI/MainUI.ts

```diff
@@ -0,0 +1,58 @@
+import CoinManagementDomain from '../../domain/CoinManagementDomain/CoinManagement';
+import ProductManagementDomain from '../../domain/ProductManagementDomain/ProductManagement';
+import PurchaseCashDomain from '../../domain/PurchaseCashDomain/PurchaseCash';
+import CoinManagementUI from './CoinManagementUI';
+import ProductManagementUI from './ProductManagementUI';
+import ProductPurchaseUI from './ProductPurchaseUI';
+import { $ } from '../../utils/dom';
+
+export default class MainUI {
+  private readonly productDomain = new ProductManagementDomain();
+  private readonly coinDomain = new CoinManagementDomain();
+  private readonly purchaseCashDomain = new PurchaseCashDomain();
+  private readonly productManagementUI = new ProductManagementUI(
+    this.productDomain,
+  );
+  private readonly coinManagementUI = new CoinManagementUI(this.coinDomain);
+  private readonly productPurchaseUI = new ProductPurchaseUI(
+    this.productDomain,
+    this.coinDomain,
+    this.purchaseCashDomain,
+  );
+
+  renderInitPage(isSignIn: boolean = false) {
+    $('#main').classList.remove('hide');
+    $('#sign-in').classList.add('hide');
+    $('#sign-up').classList.add('hide');
+    $('#user-info-edit').classList.add('hide');
+
+    if (isSignIn) {
+      this.renderProductManagementUI();
+      return;
+    }
+    this.renderProductPurchaseUI();
+  }
+
+  renderProductManagementUI() {
+    this.productManagementUI.render();
+  }
+
+  renderCoinManagementUI() {
+    this.coinManagementUI.render();
+  }
+
+  renderProductPurchaseUI() {
+    this.productPurchaseUI.render();
+  }
+
+  renderUserUI(name: string = '') {
+    this.renderUserName(name);
+    [$('.nav'), $('.signin-button'), $('.thumbnail')].forEach($button => {
+      $button.toggleAttribute('hidden');
+    });
+  }
+
+  renderUserName(name: string = '') {
+    $('.thumbnail').textContent = name.substring(0, 1);
```

Comment:
```suggestion
    $('.thumbnail').textContent = name[0]
```
문자열도 index로 접근이 가능하기에 이렇게 할수도 있긴해요~

Reply:
원래는 eslint에서 'prefer-destructuring' 관련된 에러가 발생하길래 설정하는 것보다는 string의 메서드를 사용하자고 해서 `substring`을 사용하였는데요. 오히려 한 글자만 가져오는거라 더 헷갈려 보이기도 하네요 😅
아사님이 말씀해주신 방법 적용하여 수정하였습니다 🎯 https://github.com/woowacourse/javascript-vendingmachine/pull/43/commits/d2885d1dfe32243aa957596861c435a3071faee8